### ISSUE TITLE
W1201(logging-not-lazy) complaint about string format arguments when no string format.

### ISSUE THREAD

**Issue by smoser** [PyCQA/pylint#2325]
### Steps to reproduce
```python
"""Demonstrate invalid W1201(logging-not-lazy).

Problem exists in pylint --version:
  __main__.py 2.0.0
  astroid 2.0.1
  Python 3.6.6 (default, Jun 27 2018, 14:44:17)
  [GCC 8.1.0]
"""
import logging

def mylog():
    hello, goodbye = ("hello", "goodbye")
    msg = hello + goodbye

    # it doesn't like this.
    logging.warning(hello + goodbye)

    # but these it is OK with.
    logging.warning(msg)
    logging.warning("%s%s", hello, goodbye)
````

### Current behavior
    $ pylint /tmp/show_w1201.py
    ************* Module show_w1201
    /tmp/show_w1201.py:17: [W1201(logging-not-lazy), mylog] Specify string format arguments as logging function parameters

    -----------------------------------
    Your code has been rated at 8.57/10

    shell returned 4

### Expected behavior
Complaint is technically wrong, none of these arguments contain string format operations.

### pylint --version output

    $ pylint --version
      __main__.py 2.0.0
      astroid 2.0.1
      Python 3.6.6 (default, Jun 27 2018, 14:44:17)
      [GCC 8.1.0]

**Comment by PCManticore** [comment]
Hey @smoser I think the new bug/behaviour was added during https://github.com/PyCQA/pylint/issues/2193, which was also reported by you. :) I think we should just not emit this error when a logging string is created by joining two strings, as that it's not necessarily a problem. The problem would be when having, say, something along the lines of `log.info(msg.format(...))` but string joining doesn't fall into that. If this sounds reasonable, the next course of action would be to strip that code away.

**Comment by smoser** [comment]
I knew that this was fallout of my previous actions.   :)
I think i agree with your suggestion.  I've worked around it in cloud-init, but it was just an annoyance, and I woudln't argue strongly with you if you disagreed, as I'm pretty sure that:
  log.info("%s%s", string1, string2)
is actually going to be faster due to lazy logging than:
  log.info(string1 + string2)

**Comment by Pierre-Sassoulas** [comment]
Closing as I think it's working as intended, ``logging.warning(hello + goodbye)`` cost an avoidable addition, and ``logging.warning(msg)`` should not raise as checking what is done with msg previously is not something reasonable to do for pylint.
